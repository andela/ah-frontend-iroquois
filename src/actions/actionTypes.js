const ACTION_TYPE = {
	REQUEST_LOADING: 'REQUEST_LOADING',
	SOCIAL_LOGIN_SUCCESS: 'ADD_SOCIAL_SUCCESS',
	SOCIAL_LOGIN_REDIRECT: 'ADD_SOCIAL_REDIRECT',
	LOGIN_SUCCESSFUL: 'LOGIN_USER',
	LOGIN_FAILED: 'USER_LOGIN_FAIL',
	SIGNUP_START: 'SIGNUP_START',
	SIGNUP_SUCCESS: 'SIGNUP_SUCCESS',
	SIGNUP_FAIL: 'SIGNUP_FAIL',
	SIGNUP_STOP: 'SIGNUP_STOP',
	MESSAGE: 'MESSAGE',
	LOGOUT: 'LOGOUT_USER',
	ADD_ONE_ARTICLE: 'ADD_ONE_ARTICLE',
	ADD_MANY_FROM_SERVER: 'ADD_MANY_FROM_SERVER',
	VIEW_ONE_ARTICLE: 'VIEW_ONE_ARTICLE',
	VIEW_ALL_ARTICLES: 'VIEW_ALL_ARTICLES',
	DELETE_ONE_ARTICLE: 'DELETE_ONE_ARTICLE',
	EDIT_ONE_ARTICLE: 'EDIT_ONE_ARTICLE',
	FETCH_PROFILE_FAILURE: 'FETCH_PROFILE_FAILURE',
	FETCH_PROFILE_SUCCESS: 'FETCH_PROFILE_SUCCESS',
	UPDATE_PROFILE_FAILURE: 'UPDATE_PROFILE_FAILURE',
	UPDATE_PROFILE_SUCCESS: 'UPDATE_PROFILE_SUCCESS',
	REPORT_ARTICLE_SUCCESS: 'REPORT_ARTICLE_SUCCESS',
	REPORT_ARTICLE_FAILURE: 'REPORT_ARTICLE_FAILURE',
	ADD_COMMENT: 'ADD_COMMENT',
	GET_COMMENTS: 'GET_COMMENTS',
	DELETE_COMMENT: 'DELETE_COMMENT',
	DISLIKE_ARTICLE: 'DISLIKE_ARTICLE',
	LIKE_ARTICLE: 'LIKE_ARTICLE'
};
export default ACTION_TYPE;
