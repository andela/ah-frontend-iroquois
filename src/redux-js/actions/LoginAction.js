import React from 'react';
import axios from 'axios';
import Login, {userLoginFail} from './LoginActionCreator';
import {notify} from 'react-notify-toast';
import requestLoadingAction from './request-loading';
import API_URLS, {AUTH_TOKEN} from '../../constants';

export const userLoginRequest = (user) => {
	window.store.dispatch(requestLoadingAction(true));
	return async dispatch => {
		return await axios.post(API_URLS.LOGIN_URL, {user}
		).then(response => {
			if (response.status === 200) {
				const token = response.data.user.token? response.data.user.token: null;
				localStorage.setItem(AUTH_TOKEN, token);
				window.store.dispatch(requestLoadingAction(false));
				notify.show('You have successfully logged in.','success',4000);
				window.location.assign('/');
			}
			dispatch(Login(response)
			  );

		}).catch(error => {
			dispatch(userLoginFail(error.response.data.errors));
			window.store.dispatch(requestLoadingAction(false));
            notify.show('Wrong Email/Password', 'error',4000);

		});

	};
};
