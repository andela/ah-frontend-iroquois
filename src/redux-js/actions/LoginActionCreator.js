import ACTION_TYPE from '../constants/constant';

const Login = (response) => {
	return {
		'type': ACTION_TYPE.LOGIN_SUCCESSFUL,
		'payload': response.data
	};
};

export const userLoginFail = errors => {
	return {
		'type': ACTION_TYPE.LOGIN_FAILED,
		'payload': errors
	};
};
export default Login;
