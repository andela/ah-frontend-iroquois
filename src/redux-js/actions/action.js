import requestLoadingAction from './request-loading';

const ACTIONS = {
	requestLoadingAction
};

export default ACTIONS;
