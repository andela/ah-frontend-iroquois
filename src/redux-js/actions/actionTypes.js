export const SIGNUP_START = 'SIGNUP_START';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAIL = 'SIGNUP_FAIL';
export const SIGNUP_STOP ='SIGNUP_STOP';
export const MESSAGE ='MESSAGE';
