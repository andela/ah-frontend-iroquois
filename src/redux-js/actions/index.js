import axios from 'axios'
import {
    passwordInvokeSuccess,
    resetPasswordError,
    resetPasswordSuccess
} from './action';
import {notify} from 'react-notify-toast';
import {invokeResetPasswordUrl, passwordRedirectUrl, resetPasswordUrl} from '../constants/constant';
import requestLoadingAction from './request-loading';

const NotificationSuccess = (message) => {
	return notify.show(message, "success", 5000);
};

export const NotificationError = (message) => {
	return notify.show(message, "error", 5000);
};


export const PasswordInvokeThunk = (email) => {
    return (dispatch) => {
		dispatch(requestLoadingAction(true));

        const userData = {
            user: {
                email: email,
                call_back: passwordRedirectUrl
            }
        };
        return axios.post(invokeResetPasswordUrl, userData)
			.then(response => {

                dispatch(passwordInvokeSuccess(response.data.user.message));
				NotificationSuccess(response.data.user.message);
				dispatch(requestLoadingAction(false));
            }).catch(error => {
                // Give a positive message to the user even when email is not in database.
				dispatch(passwordInvokeSuccess("Check your email for a link"));
				NotificationSuccess("Check your email for a link");
				dispatch(requestLoadingAction(false));
        })
    }
};

const asyncFunc = (dispatch, userData, myHeader) => {

	return axios.put(resetPasswordUrl, userData, myHeader)
		.then(response => {
			dispatch(requestLoadingAction(false));

			dispatch(resetPasswordSuccess("Password reset successfully continue to login"));
			NotificationSuccess("Password reset successfully continue to login");

		}).catch(error => {

			dispatch(requestLoadingAction(false));

			if (error.response.data.user){
				dispatch(resetPasswordError("Invalid toke, Please get a fresh password reset link"));
				NotificationError("Invalid toke, Please get a fresh password reset link");
			}
		})
};

export const ResetPasswordThunk = (new_password, confirm_password, token) => {
    return (dispatch) => {

		if(new_password.trim() !== confirm_password.trim()){
			dispatch(resetPasswordError("Oops!! passwords don't match"));
			NotificationError("Oops!! passwords don't match");
			return;
		}

		// check that email contains both letters and numbers
		if (!(new_password.match(/\d/) && new_password.match(/[a-z]/i))){
			dispatch(resetPasswordError("Password should contain at least one letter and number"));
			return;
		}


		const userData = {
            user: {
                password: new_password
            }
        };

        let myHeader = {headers: {"Content-Type": "application/json", "Accept": "application/json", "Authorization": "Token " + token}};

		dispatch(requestLoadingAction(true));

		// Make calls to the API
		return asyncFunc(dispatch, userData, myHeader)
    };

};




