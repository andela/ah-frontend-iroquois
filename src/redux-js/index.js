import ACTIONS from './actions/action';
import { store } from './stores';

window.store = store;
window.requestLoadingAction = ACTIONS.requestLoadingAction;
window.socialLoginServiceAction = ACTIONS.socialLoginServiceAction;
