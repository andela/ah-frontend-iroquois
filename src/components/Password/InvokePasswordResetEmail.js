import React from 'react'
import {connect} from 'react-redux'
import {PasswordInvokeThunk} from "../../redux-js/actions";
import SubmitContent from './common';
import {emailValidation, validateEmail} from '../Utils/utils';

class InvokePasswordResetEmail extends React.Component{
	constructor(props){
		super(props);
		this.state = {
			emailHasError: true,
			emailError: "",
			email: ""
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(evt) {
		this.setState({[evt.target.name]: evt.target.value});
		if (evt.target.name === 'email'){
			validateEmail(evt.target.value, this)
		}
	}

	handleSubmit = (event) => {
		event.preventDefault();
		this.props.dispatch(PasswordInvokeThunk(this.state.email));
	};

	emailForm = () => {
		return <form onSubmit={this.handleSubmit}>
			<div className="row">
				<div className="input-field col s12 m12">
					<i className="material-icons prefix">email</i>
					<input id="email" type="email" className="validate" name={'email'} onChange={this.handleChange}/>
					<label htmlFor="email">Email</label>
					<span className={'red-text'}>{this.state.emailError}</span>
				</div>
			</div>
			{SubmitContent(false, this.state.emailHasError)}
		</form>
	};

    render(){
    	return <div className="row">
			<div className="col s12 m6 offset-m3 l4 offset-l4">
				<div className="card white">
					<div className="card-content black-text">
						<div className={'row'}>
							<div className={'col m10 offset-m1'}><span className="card-title teal-text"><h5>Get password reset link</h5></span>
								<p className={'grey-text'}>Forgotten your password? Enter your email address below, and weâ€™ll send instructions for setting a new one.</p>
							</div>
						</div>
						<div className={'row'}>
							<div className={'col s12 m10 offset-m1'}>
								{this.emailForm()}
							</div>
						</div>
					</div>
				</div>

			</div>


        </div>
    }
}

const mapStateToProps = state => ({data: state });

export default connect(mapStateToProps)(InvokePasswordResetEmail)
