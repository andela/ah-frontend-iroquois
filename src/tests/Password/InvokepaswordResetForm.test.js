import React from 'react';
import InvokePasswordResetEmail from "../../components/Password/InvokePasswordResetEmail";
import {BrowserRouter, MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import conf from 'redux-mock-store';

import {PasswordInvokeThunk} from '../../redux-js/actions';

const props = {
	data: {
		PasswordReducer: {
			invokePasswordErrorMessage: "",
			invokePasswordSuccessMessage: ""
		}
	}
};

const store = conf([thunk])({});

describe("Check loader has class", () => {

	it("renders correctly", () => {
		const tree = renderer.create(<BrowserRouter><Provider store={store}><InvokePasswordResetEmail {...props}/></Provider></BrowserRouter>).toJSON();
		expect(tree).toMatchSnapshot();

		mount(
			<MemoryRouter>
			<Provider store={store}>
			<InvokePasswordResetEmail {...props}/>
			</Provider>
		</MemoryRouter>);
	});


});

describe('handle Invoke email for password reset', () => {
    let component;

    beforeEach(() => {
		component = mount(<MemoryRouter>
			<Provider store={store}>
				<InvokePasswordResetEmail dispatch={PasswordInvokeThunk("email@g.com")}/>
			</Provider>
		</MemoryRouter>);

    });


	it('should submit user email without failing', () => {

		component.find('input#email').simulate('change', {target: {name: 'email', value: 'ewrfg'}});
		component.find('input#email').simulate('change', {target: {name: 'email', value: ''}});
		component.find('input#email').simulate('change', {target: {name: 'emailEr', value: ''}});
		component.find('input#email').simulate('change', {target: {name: 'email', value: 'sdfg@dfg'}});

		component.find('form').simulate('submit');

	})

});


