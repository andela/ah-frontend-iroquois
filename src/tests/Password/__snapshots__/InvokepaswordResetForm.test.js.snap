// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check loader has class renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m6 offset-m3 l4 offset-l4"
  >
    <div
      className="card white"
    >
      <div
        className="card-content black-text"
      >
        <div
          className="row"
        >
          <div
            className="col m10 offset-m1"
          >
            <span
              className="card-title teal-text"
            >
              <h5>
                Get password reset link
              </h5>
            </span>
            <p
              className="grey-text"
            >
              Forgotten your password? Enter your email address below, and weâ€™ll send instructions for setting a new one.
            </p>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col s12 m10 offset-m1"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <i
                    className="material-icons prefix"
                  >
                    email
                  </i>
                  <input
                    className="validate"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    type="email"
                  />
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <span
                    className="red-text"
                  >
                    
                  </span>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s4 m6 l6"
                >
                  <button
                    className="btn btn-small"
                    disabled={true}
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
                <div
                  className="input-field col s8 m6 l6"
                >
                  <button
                    className="btn btn-small white black-text"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Cancel
                    </a>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
